drivers/flash/spi_nor.c

```diff
@@ -540,6 +540,45 @@ static int spi_nor_wrsr(const struct device *dev,
 	return ret;
 }
 
+#define MX25R_HIGH_PERF_MODE BIT(1)
+
+#ifdef CONFIG_SPI_NOR_MX_HIGH_PERF_MODE
+static int mx_specific_enable_high_perf_mode(const struct device *dev, uint8_t jedec_id[])
```

Comment:
I'm not actually convinced that this is the best approach for providing such feature. This way it is only possible to activate the high performance mode permanently and for all instances of the flash chip (I can imagine that one may want to use a fancy setup with two flash chips and the high performance mode only for one of them).
How about providing such function as a custom API of the spi_nor driver? Then it could be called on demand.
Something similar to what we've got for the nrf_qspi_nor driver:
https://github.com/zephyrproject-rtos/zephyr/blob/b953ff1418d2490d2a030f2a0cac771de3ae4128/include/zephyr/drivers/flash/nrf_qspi_nor.h#L15-L32

Reply:
This is valid argument. Shouldn't we then consider adding DTS entry that will allow to enable the feature per device?

Reply:
Using device tree sounds like a nice solution to the multiple flash chips problem. I can add a new property for enabling high performance mode in the PR.

I'm not sure about the custom API. Perhaps you can explain a bit more about this? How would this be used from the application code? The custom API would have to be protected with #ifdefs or device-tree macros, right? Is it really necessary to have it as part of this PR, or could it be a future improvement?

Reply:
I'm not very familiar with driver work, but how can it support multiple instances when the driver only handles device instance number 0?

```c
DEVICE_DT_INST_DEFINE(0, &spi_nor_init, NULL,
		 &spi_nor_data_0, &spi_nor_config_0,
		 POST_KERNEL, CONFIG_SPI_NOR_INIT_PRIORITY,
		 &spi_nor_api);
```

Reply:
Right, the driver currently supports only one instance. I haven't checked that before adding my comment. From looking at the dts nodes I had the impression that multiple instances are possible (well, maybe in some future the driver could be extended with such support). But still, I think that an option to enable the high performance mode fits better in [jedec,spi-nor-common.yaml](https://github.com/zephyrproject-rtos/zephyr/blob/0c2607786dd2f24cee8b215c7bb31bf4d6eabaed/dts/bindings/mtd/jedec%2Cspi-nor-common.yaml) than in Kconfig.nor (and it could be then utilized by the nrf_qspi_nor driver as well).
The custom API was just a proposal. Indeed, it could be added as an improvement in the future.

Reply:
I've updated the PR now with a custom binding for the Macronix Ultra Low Power series that define a property to enable high performance mode. Had a chat with @mbolivar, and he recommended not using spi-nor-common when the property is only for one series from Macronix, so I added a new binding for `mxicy,mx25r-spi` which use the existing spi-nor driver. I chose `mxicy` for Macronix, since [that's what is used in the Linux repo](https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/mtd/mxicy%2Cnand-ecc-engine.yaml).
Removed the kconfig, as I don't think it's necessary anymore.